core/dbt/events/base_types.py

```diff
@@ -42,31 +30,28 @@ def __post_init__(self):
 
 
 # TODO add exhaustiveness checking for subclasses
-# can't use ABCs with @dataclass because of https://github.com/python/mypy/issues/5374
 # top-level superclass for all events
-class Event(metaclass=ABCMeta):
+@dataclass
+class Event(dbtClassEventMixin, metaclass=ABCMeta):
     # fields that should be on all events with their default implementations
     log_version: int = 1
     ts: Optional[datetime] = None  # use getter for non-optional
     ts_rfc3339: Optional[str] = None  # use getter for non-optional
     pid: Optional[int] = None  # use getter for non-optional
-    node_info: Optional[Node]
 
     # four digit string code that uniquely identifies this type of event
     # uniqueness and valid characters are enforced by tests
-    @abstractproperty
+    @property
```

Comment:
This is an issue. Removing it as abstract means mypy can no longer detect that events are missing codes. You can see this by commenting out a code field in types.py and running mypy.

Reply:
But if we leave it as an abstractproperty then we can't make it a dataclass, which mashumaro requires.

Reply:
Checking for a code could be tested instead of relying on mypy.

Reply:
If this is the case, I don't think Mashumaro is a good fit for serialization in this module.

Reply:
Is there a way we could make something lower on the hierarchy inherit from the serialization class? like the loglevel classes maybe?